schema/field.go

```diff
@@ -94,17 +94,19 @@ type FieldHandler func(ctx context.Context, value interface{}, params map[string
 // value to validate as argument and returned the normalized value or an error
 // if validation failed.
 type FieldValidator interface {
-	Validate(value interface{}) (interface{}, error)
+	// Validate a field with a value. With enforce=false this function only converts to internal type.
+	// enforce=true activates all other per type checks
+	Validate(value interface{}, enforce bool) (interface{}, error)
```

Comment:
This really isn't the right interface to allow for query operator validation.
1. It does two different things based on a Boolean value -- design-wise this means it should be two methods, not one, at the very least at the public interface level.
2. It doesn't really handle all the cases it needs to handle for operator value validation; It _doesn't_ only need to validate that the value is correct, it also needs to check that the operator itself makes sense for the given type. E.g. `$regex` don't make sense for time or numbers, and `$gt` does not make sense for strings.

This is why the suggested interface in #170 includes an optional interface that takes both the operator and the value as input parameters. Perhaps strings for operators isn't the best choice though, and maybe simialr operations can be grouped. E.g. if `$eq` is allowed, `$ne` should be allowed, if `$gt` is allowed, `$lt` should be allowed"

```go
type QueryOp uint

const (
    QueryOpEquality = iota // allow $eq, $ne
    QueryOpLessGreater // allow $lt, $eq
    QueryOpIn // allow $in $nin
    QueryOpRegex // allow $regex
)

type OperatorValidator interface {
    // Return an error if op isn't allowed, or if value isn't of an acceptable type or format.
    ValidateOperator(op QueryOp, value interface{}) (interface{}, error)
}
```

Reply:
Hmm, I just did the following with the patch above:
```
GET http://localhost:3001/api/visits?filter={name: {$regex: 5 }}
```
I am getting this result:
```
{
  "code": 422,
  "issues": {
    "filter": [
      "char 16: name: $regex: not a string"
    ]
  },
  "message": "URL parameters contain error(s)"
}
```

So I think matching types are covered. Also Since Mongo allows using string lexical comparison, why we don't allow it as well? 
Let me try polish some more.